src/libraries/System.Private.CoreLib/src/System/Buffers/Binary/ReaderLittleEndian.cs

```diff
@@ -25,6 +25,23 @@ public static double ReadDoubleLittleEndian(ReadOnlySpan<byte> source)
                 MemoryMarshal.Read<double>(source);
         }
 
+        /// <summary>
+        /// Reads a <see cref="Half" /> from the beginning of a read-only span of bytes, as little endian.
+        /// </summary>
+        /// <param name="source">The read-only span to read.</param>
+        /// <returns>The little endian value.</returns>
+        /// <remarks>Reads exactly 2 bytes from the beginning of the span.</remarks>
+        /// <exception cref="ArgumentOutOfRangeException">
+        /// <paramref name="source"/> is too small to contain a <see cref="Half" />.
+        /// </exception>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static Half ReadHalfLittleEndian(ReadOnlySpan<byte> source)
+        {
+            return !BitConverter.IsLittleEndian ?
+                BitConverter.Int16BitsToHalf(ReverseEndianness(MemoryMarshal.Read<short>(source))) :
+                MemoryMarshal.Read<Half>(source);
```

Comment:
pedantic nit: removing extra `!` can simplify the code?

```suggestion
            return BitConverter.IsLittleEndian ?
                MemoryMarshal.Read<Half>(source) :
                BitConverter.Int16BitsToHalf(ReverseEndianness(MemoryMarshal.Read<short>(source)));
```

Reply:
I just copied the code body from `double` overload and changed the types. They should be consistent after this PR right?